app/controllers/apps_controller.rb

```diff
@@ -40,6 +38,13 @@ def destroy
   end
   private
 
+  def app_owner?
+    if !App.find(params[:id]).owners.include?(current_user)
```

Comment:
could you extract this logic out of the controller somehow? this is a lot for the controller to know about!